eth/ethconfig/config.go

```diff
@@ -167,10 +167,8 @@ func CreateConsensusEngine(config *params.ChainConfig, db ethdb.Database) (conse
 	if config.Taiko {
 		return taiko.New(config, db), nil
 	}
-	// Geth v1.14.0 dropped support for non-merged networks in any consensus
-	// mode. If such a network is requested, reject startup.
-	if !config.TerminalTotalDifficultyPassed {
-		return nil, errors.New("only PoS networks are supported, please transition old ones with Geth v1.13.x")
+	if config.TerminalTotalDifficulty == nil {
+		return nil, fmt.Errorf("only PoS networks are supported, please transition old ones with Geth v1.13.x")
```

Comment:
```suggestion
		log.Error("Geth only supports PoS networks. Please transition legacy networks using Geth v1.13.x.")
		return nil, fmt.Errorf("'terminalTotalDifficulty' is not set in genesis block")
```

Let's keep it same with the upstream: https://github.com/ethereum/go-ethereum/blob/master/eth/ethconfig/config.go#L162